mount/infoParser.go

```diff
@@ -0,0 +1,1037 @@
+// 3074 2712 0:153 / /usr/src/linux-headers-5.4.0-62 ro,relatime - shiftfs /usr/src/linux-headers-5.4.0-62 rw
+//
+// Example 2: mountID-3074 is a 'self' mount of the original mountID-2706 entry.
+//
+// 2706 2192 0:155 /resolv.conf /etc/resolv.conf rw,relatime - shiftfs /var/lib/docker/containers... rw
+// 3074 2706 0:155 /resolv.conf /etc/resolv.conf rw,relatime - shiftfs /var/lib/docker/containers... rw
+//
+func (mi *mountInfoParser) IsSelfMount(info *domain.MountInfo) bool {
+	if info == nil {
+		return false
+	}
+
+	infoParent := mi.GetParentMount(info)
+	if infoParent == nil {
+		return false
+	}
+
+	return info.Root == infoParent.Root &&
+		info.MountPoint == infoParent.MountPoint &&
+		info.Source == infoParent.Source
+}
+
+// IsOverlapMount determines if the mountpoint associated to a mountInfo entry
+// overlaps with any other mountpoint in the mountInfo tree. A 'self' mount
+// is a special case (subset) of an 'overlap' one.
+//
+// Example 1: Same as as the above one (IsSelfMount method).
+//
+// 2712 2192 0:153 / /usr/src/linux-headers-5.4.0-62 ro,relatime - shiftfs /usr/src/linux-headers-5.4.0-62 rw
+// 3074 2712 0:153 / /usr/src/linux-headers-5.4.0-62 ro,relatime - shiftfs /usr/src/linux-headers-5.4.0-62 rw
+//
+// Example 2: mountID-3074 is an 'overlap' mount of the original mountID-2706 entry ]
+//
+// 2706 2192 0:155 /resolv.conf /etc/resolv.conf rw,relatime - shiftfs /var/lib/docker/containers... rw
+// 3074 2706 0:6 /null /etc/resolv.conf rw,nosuid,noexec,relatime master:2 - devtmpfs udev rw,size=4048120k,nr_inodes=1012030,mode=755
+//
+func (mi *mountInfoParser) IsOverlapMount(info *domain.MountInfo) bool {
+
+	if info == nil {
+		return false
+	}
+
+	infoParent := mi.GetParentMount(info)
+	if infoParent == nil {
+		return false
+	}
+
+	return info.MountPoint == infoParent.MountPoint
+}
+
+// IsBindMount verifies if the passed mountinfo entry is a bind-mount candidate.
+func (mi *mountInfoParser) IsBindMount(info *domain.MountInfo) bool {
+
+	if info == nil {
+		return false
+	}
+
+	mh := mi.service.mh
+	if mh == nil {
+		return false
+	}
+
+	devIdSlice := mi.devIdInfo[info.MajorMinorVer]
+
+	for _, elem := range devIdSlice {
+		if elem.MountID == info.MountID {
+			continue
+		}
+
+		if elem.Root == info.Root && elem.Source == info.Source {
+			return true
+		}
+	}
+
+	return false
+}
+
+// IsBindMount verifies if the passed mountinfo entry is a read-only bind-mount.
+func (mi *mountInfoParser) IsRoBindMount(info *domain.MountInfo) bool {
+
+	if info == nil {
+		return false
+	}
+
+	mh := mi.service.mh
+	if mh == nil {
+		return false
+	}
+
+	devIdSlice := mi.devIdInfo[info.MajorMinorVer]
+
+	for _, elem := range devIdSlice {
+		if elem.MountID == info.MountID {
+			continue
+		}
+
+		if elem.Root == info.Root && elem.Source == info.Source {
+			return mh.StringToFlags(elem.Options)&unix.MS_RDONLY == unix.MS_RDONLY
+		}
+	}
+
+	return false
+}
+
+func (mi *mountInfoParser) equivalentMounts(m1, m2 *domain.MountInfo) bool {
```

Comment:
Disregard this function (equivalentMounts) and the next one (equivalentMountAncestors) as I didn't use them at the end. Will remove them.